src/librustc_mir/interpret/snapshot.rs

```diff
@@ -383,28 +437,29 @@ impl<'a, 'mir, 'tcx, M> Hash for EvalSnapshot<'a, 'mir, 'tcx, M>
 
 // Not using the macro because we need special handling for `memory`, which the macro
 // does not support at the same time as the extra bounds on the type.
-impl<'a, 'b, 'mir, 'tcx, M> HashStable<StableHashingContext<'b>>
-    for EvalSnapshot<'a, 'mir, 'tcx, M>
-    where M: Machine<'mir, 'tcx>,
+impl<'a, 'b, 'mir, 'tcx> HashStable<StableHashingContext<'b>>
+    for EvalSnapshot<'a, 'mir, 'tcx>
 {
     fn hash_stable<W: StableHasherResult>(
         &self,
         hcx: &mut StableHashingContext<'b>,
         hasher: &mut StableHasher<W>)
     {
+        // Not hashing memory: Avoid hashing memory all the time during execution
```

Comment:
You only need `HashStable` if you're putting these snapshots through queries (because incremental needs to save/reload them) - is that the case?

Reply:
I have no idea. I didn't write this code and I do not understand why they are using stable hashing. See https://github.com/rust-lang/rust/issues/54384 and ask @oli-obk ;)

The purpose of this PR is mostly to get the loop detector out of my way, so that my next PR does not have to be concerned with this so much.